subsys/cpp/cpp_new.cpp

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2018
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <new>
+#include <kernel.h>
+
+void* operator new(size_t count)
+{
+#if (CONFIG_HEAP_MEM_POOL_SIZE > 0)
+	void* ptr = k_malloc(count);
+#if defined(__cpp_exceptions) && defined(CONFIG_LIB_CPLUSPLUS)
+	if (!ptr)
+		throw std::bad_alloc();
+#endif
+	return ptr;
+#else
+	(void)count;
```

Comment:
ARG_UNUSED() is the standard macro for this.   If this were a header that didn't want to suck in zephyr dependencies, I think it would be fine.  But you have kernel.h included above.